addons/project/models/project_project.py

```diff
@@ -186,7 +186,7 @@ def _read_group_stage_ids(self, stages, domain, order):
 
     # Not `required` since this is an option to enable in project settings.
     stage_id = fields.Many2one('project.project.stage', string='Stage', ondelete='restrict', groups="project.group_project_stages",
-        tracking=True, index=True, copy=False, default=_default_stage_id, group_expand='_read_group_stage_ids')
+        tracking=True, index=True, copy=False, default=_default_stage_id, group_expand='_read_group_stage_ids', domain="[('company_id', 'in', [*allowed_company_ids, False])]")
```

Comment:
My bad, I think this is sliiiightly better:
```suggestion
        tracking=True, index=True, copy=False, default=_default_stage_id, group_expand='_read_group_stage_ids', domain="[('company_id', 'in', (*allowed_company_ids, False))]")
```
(I wouldn't have highlighted it if it was the rest was good to go)